typed-racket-test/succeed/flonum-typecheck.rkt

```diff
@@ -0,0 +1,123 @@
+#lang typed/racket
+(require racket/flonum
+         typed/rackunit)
+
+
+;;Flonum typechecking tests
+;; -FlZero -Fl
+(check-eq?
```

Comment:
I'm confused at what these cases are testing. It seems like this issue was about filters but this is testing that in a very indirect manner, by making sure that the optimizer doesn't do dead code elimination on the wrong branch. Would these be better tested by the unit tests for typechecking?